qiskit/_parallel.py

```diff
@@ -0,0 +1,119 @@
+#
+#    2. Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in the
+#       documentation and/or other materials provided with the distribution.
+#
+#    3. Neither the name of the QuTiP: Quantum Toolbox in Python nor the names
+#       of its contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+#    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+#    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#    HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#    SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#    LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#    DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#    THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#    (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+###############################################################################
+
+"""Routines for running Python functions in parallel using process pools
+from the multiprocessing library.
+"""
+
+import os
+import platform
+from multiprocessing import Pool
+from qiskit._qiskiterror import QISKitError
+from qiskit._util import local_hardware_info
+
+# Number of local physical cpus
+CPU_COUNT = local_hardware_info()['cpus']
+
+
+def parallel_map(task, values, task_args=tuple(), task_kwargs={},  # pylint: disable=W0102
+                 num_processes=CPU_COUNT):
+    """
+    Parallel execution of a mapping of `values` to the function `task`. This
+    is functionally equivalent to::
+        result = [task(value, *task_args, **task_kwargs) for value in values]
+
+    On Windows this function defaults to a serial implimentation to avoid the
+    overhead from spawning processes in Windows.
+
+    Parameters:
+        task (func): Function that is to be called for each value in ``task_vec``.
+        values (array_like): List or array of values for which the ``task``
+                            function is to be evaluated.
+        task_args (list): Optional additional arguments to the ``task`` function.
+        task_kwargs (dict): Optional additional keyword argument to the ``task`` function.
+        num_processes (int): Number of processes to spawn.
+
+    Returns:
+        result: The result list contains the value of
+                ``task(value, *task_args, **task_kwargs)`` for
+                    each value in ``values``.
+
+    Raises:
+        QISKitError: If user interupts via keyboard.
+    """
+
+    def _callback(x):  # pylint: disable=W0613
```

Comment:
It was left there for @atilag to use later when he implements his sub/pub idea.  I can remove it if you do not want the stub there.

Reply:
In that case, yes, please - it can be reintroduced when it does really have a purpose, as currently it hinders readability a bit without real benefit.

Reply:
Ok